File: Backend/models/pokemon.go
package models

type Deck struct {
	Id    string
	Name  string
	Type  string
	Image string
	Cards []string
}

type DeckList struct {
	DeckId []string
}

type ApiPokemonResponse struct {
	Data []Card
}
type Card struct {
	Id          string
	Name        string
	Images      CardImages // This comes from images.large in the card model in the API
	EvolvesFrom string
}

type CardImages struct {
	Large string
}

File: Frontend/src/models/pokemon.ts

export interface IDeckCard {
	Id: string;
	Name: string;
	Image: string; // This comes from images.large in the card model in the API
	Type: string;
}

export interface Deck {
	Id:    string;
	Name:  string;
	Type:  string;
	Image: string;
	Cards: IDeckCard[];
}
